Enable binding to Canvas attached properties

wieslawsoltes opened the pull request:
But I still can't get working fully. 

When I change bound property and raise INPC the e.g. `Canvas.Left` attached property is not affected.

WPF is using callback to invalidate UI element manually `OnPositioningChanged` http://referencesource.microsoft.com/#PresentationFramework/src/Framework/System/Windows/Controls/Canvas.cs,214

Update:

I have managed to make it working by observing property changes and calling `InvalidateArrange` manually on Parent.

Commit:
Changed attached property owner to Canvas

Commit:
Added Canvas properties

Commit:
Set default BindingMode for attached properties

Commit:
Observe property changes and call InvalidateArrange manually on Parent

grokys commented:
Sorry for the delay in looking at this - i need to take a look at what is causing the elements to not get invalidated. Will try to work it out soon.

wieslawsoltes commented:
No problem. After you will resolve the invalidated issue I would like to investigate binding to properties for other controls too. Also I have noticed that binding to attached inside `DataTemplate` of `ItemsControl` does not work.

amerkoleci commented:
Can we have Canvas property merged?

grokys commented:
I've commited the fix for the canvas property owner separately.

wieslawsoltes commented:
@grokys Any progress on issue why canvas is not invalidated.

grokys commented:
Yeah, your PR was just about right - `AffectsArrange` causes `InvalidateArrange` to be called *on the control whose property has changed, not on the parent`. I fixed it in [another commit](https://github.com/Perspex/Perspex/commit/7a67889b45bb04f423e5db3e0f95c1be0273fb55) rather than using this PR because there seem to be unrelated changes here (CLR accessors for `Left`, `Top` etc) which I don't understand...

wieslawsoltes commented:
@grokys I added CLR properties as this was needed to get things working, after your fixed all works as expected. I suspect this was related to attached properties issues I recently and this was also fixed.
